beamer/agent.py

```diff
@@ -84,37 +97,41 @@ def _init(self) -> None:
 
         self._event_processor = EventProcessor(self.context)
 
-        self._event_monitor_l2a = EventMonitor(
-            web3=w3_l2a,
-            contracts=(request_manager,),
-            deployment_block=l2a_contracts_info["RequestManager"].deployment_block,
-            on_new_events=[self._event_processor.add_events],
-            on_sync_done=self._event_processor.mark_sync_done,
-            poll_period=POLL_PERIOD,
-        )
-
-        self._event_monitor_l2b = EventMonitor(
-            web3=w3_l2b,
-            contracts=(fill_manager,),
-            deployment_block=l2b_contracts_info["FillManager"].deployment_block,
-            on_new_events=[self._event_processor.add_events],
-            on_sync_done=self._event_processor.mark_sync_done,
-            poll_period=POLL_PERIOD,
-        )
+        self._event_monitors: Dict[ChainId, EventMonitor] = {}
+
+        for chain_id in chain_ids:
+            if chain_id == source_chain_id:
+                self._event_monitors[chain_id] = EventMonitor(
+                    web3=w3_source,
+                    contracts=(source_request_manager, source_fill_manager),
+                    deployment_block=_get_deployment_block(source_contracts_info),
+                    on_new_events=[self._event_processor.add_events],
+                    on_sync_done=self._event_processor.mark_sync_done,
+                    poll_period=POLL_PERIOD,
+                )
+            else:
+                self._event_monitors[chain_id] = EventMonitor(
+                    web3=w3_target,
+                    contracts=(target_request_manager, target_fill_manager),
+                    deployment_block=_get_deployment_block(target_contracts_info),
+                    on_new_events=[self._event_processor.add_events],
+                    on_sync_done=self._event_processor.mark_sync_done,
+                    poll_period=POLL_PERIOD,
+                )
```

Comment:
minor nitpick: you could pull out the creation of the EventMonitor and assignment in the dict to after the if clause, and assign the source and target rollup dependent variables inside of the if/else